app/controllers/people/omniauth_callbacks_controller.rb

```diff
@@ -55,17 +72,29 @@ def signup
   ###########
 
   def handle_new_person
-    redirect_to new_group_member_path(signup_mode: @service,
-                                      group_id: current_group.id,
-                                      person: { oauth: encrypt_token(@auth) })
+    case @signup_reason
```

Comment:
for later: this is starting to get pretty deeply branched. if it keeps getting that way, let's consider introducing a helper concern something like: https://gist.github.com/aguestuser/541ef99a90aa405036da21a7e6fd8092 ?